.github/scripts/run_qemu_tests.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env sh
+
+# https://gist.github.com/mvidner/8939289 - qemu characters
+# TODO: implement the JSON version
+# https://www.qemu.org/docs/master/devel/qapi-code-gen.html
+
+qemu-system-i386 -cdrom bin/ArtOS.iso -serial file:serial.log -boot a -s -m 2G -no-reboot -nographic -monitor unix:qemu-monitor-socket,server,nowait&
+sleep 5
```

Comment:
These sleeps should instead somehow cleverly poll the serial log (I could write an "expect" function and run this whole thing in python instead to make it legible.

Reply:
If you had a standard logging format, you could read the serial log file until you see an init entry for that part of the system. Still think you need to switch from grub though :wink:  Bootloader? More like shitloader.